scalatest/src/main/scala/org/scalatest/RecoverMethods.scala

```diff
@@ -188,13 +188,18 @@ trait RecoverMethods {
   def recoverToSucceededIf[T <: AnyRef](future: Future[Any])(implicit classTag: ClassTag[T], exCtx: ExecutionContext, pos: source.Position): Future[Assertion] = {
     val clazz = classTag.runtimeClass
     future.failed.transform(
-      ex =>
+      rawEx => {
+        val ex =
+          rawEx match {
+            case execEx: java.util.concurrent.ExecutionException => execEx.getCause
+            case other => other
+          }
```

Comment:
@cheeseng What was this change for? I can see you're grabbing the cause if it is an execution exception, but what was the situation in which this bug showed up?